File: Product_Management/src/main/java/com/example/Product_Management/service/ProductService.java
package com.example.Product_Management.service;

import com.example.Product_Management.model.Product;
import org.springframework.stereotype.Component;


import java.util.*;

@Component
public interface ProductService {
    public Product saveProduct(Product product);
    public List<Product> getAllProduct();
    public Product getProductById(Integer id);
    public String deleteProduct(Integer id);
    public Product editProduct(Product product, Integer id);
}

File: product/src/service/product.service.js
import axios from "axios";

const API_URL="http://localhost:8080"
class ProductService {      
  saveProduct(product) {
    return axios.post(API_URL + "/saveProduct", product);
  }
  getAllProducts() {
    return axios.get(API_URL + "/");
  }
  getProductById(id) {
    return axios.get(API_URL + "/"+id);
  }
  editProduct(product) {
    return axios.put(API_URL + "/editProduct/" + product.id, product);
  }
  deleteProduct(id) {
    return axios.get(API_URL + "/deleteProduct/"+id);
  }
}
export default new ProductService;